federation/pkg/kubefed/init/init.go

```diff
@@ -442,7 +454,16 @@ func createNamespace(clientset client.Interface, federationName, namespace strin
 	return clientset.Core().Namespaces().Create(ns)
 }
 
-func createService(cmdOut io.Writer, clientset client.Interface, namespace, svcName, federationName, apiserverAdvertiseAddress string, apiserverServiceType v1.ServiceType, dryRun bool) (*api.Service, []string, []string, error) {
+func createService(cmdOut io.Writer, clientset client.Interface, namespace, svcName, federationName, apiserverAdvertiseAddress string, apiserverPort int32, apiserverServiceType v1.ServiceType, dryRun bool) (*api.Service, []string, []string, error) {
+	port := api.ServicePort {
+			Name:       "https",
+			Protocol:   "TCP",
+			Port:       443,
+			TargetPort: intstr.FromString(apiServerSecurePortName),
+	}
+	if apiserverServiceType == v1.ServiceTypeNodePort {
+		port.NodePort = apiserverPort
```

Comment:
But if users configure `--api-server-port=0`, what will happen here? What I see is a zero value has been given to `NodePort`. kubernetes will assign a random port in line 486, right? I might miss something here :)